kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/Constraints.java

```diff
@@ -448,12 +448,26 @@ public String toString() {
 
   public Map<String, String> toQueryMap() {
     Map<String, String> query = Maps.newLinkedHashMap();
+    return toQueryMap(query);
+  }
+
+  /**
+   * Get a normalized query map for the constraints. A normalized query map will
+   * be equal in value and iteration order for any logically equivalent set of
+   * constraints.
+   */
+  public Map<String, String> toNormalizedQueryMap() {
+    Map<String, String> query = Maps.newTreeMap();
+    return toQueryMap(query);
+  }
+
+  private Map<String, String> toQueryMap(Map<String, String> queryMap) {
     for (Map.Entry<String, Predicate> entry : constraints.entrySet()) {
       String name = entry.getKey();
       Schema fieldSchema = SchemaUtil.fieldSchema(schema, strategy, name);
-      query.put(name, Predicates.toString(entry.getValue(), fieldSchema));
+      queryMap.put(name, Predicates.toNormalizedString(entry.getValue(), fieldSchema));
```

Comment:
I should have thought about it a bit more when I switched it over. The addition of the flag to normalize the predicate (or not) was minor enough that it still reduced the amount of code. Updated in the latest push.